File: CV_tasks/Cpp/image_manipulation.cpp
#include <iostream>
#include <opencv2/opencv.hpp>

int main() {
    cv::Mat image = cv::imread("/Users/milena/Desktop/images/test.png");

    if (image.empty()) {
        std::cerr << "Error: Couldn't load the image." << std::endl;
        return -1;
    }

    //---resizing---
    cv::Mat resized_image;
    cv::resize(image, resized_image, cv::Size(400, 300));

    cv::imshow("Resized Image", resized_image);
    cv::waitKey(0);
    cv::destroyAllWindows();

    cv::imwrite("resized_image.jpg", resized_image);

    //flipping
    // cv::Mat flipped_vertical;
    // cv::flip(image, flipped_vertical, 0);

    // cv::imshow("Flipped Vertical Image", flipped_vertical);
    // cv::waitKey(0);
    // cv::destroyAllWindows();

    // cv::imwrite("flipped_vertical_image.jpg", flipped_vertical);

    // cv::Mat flipped_horizontal;
    // cv::flip(image, flipped_horizontal, 1);

    // cv::imshow("Flipped Horizontal Image", flipped_horizontal);
    // cv::waitKey(0);
    // cv::destroyAllWindows();

    // cv::imwrite("flipped_horizontal_image.jpg", flipped_horizontal);


    //cropping

    //rotating

   

    return 0;
}

File: CV_tasks/py/image_manipulation.py
import cv2

# Load an image
image = cv2.imread('images/test.webp')

if image is None:
    print("Error: Couldn't load the image.")
    exit(-1)

# Resizing
resized_image = cv2.resize(image, (400, 300))
cv2.imshow("Resized Image", resized_image)
cv2.waitKey(0)
cv2.destroyAllWindows()
cv2.imwrite("resized_image.jpg", resized_image)

# Flipping vertical
flipped_vertical = cv2.flip(image, 0)
cv2.imshow("Flipped Vertical Image", flipped_vertical)
cv2.waitKey(0)
cv2.destroyAllWindows()
cv2.imwrite("flipped_vertical_image.jpg", flipped_vertical)

# Flipping horizontal
flipped_horizontal = cv2.flip(image, 1)
cv2.imshow("Flipped Horizontal Image", flipped_horizontal)
cv2.waitKey(0)
cv2.destroyAllWindows()
cv2.imwrite("flipped_horizontal_image.jpg", flipped_horizontal)

# Cropping
x, y, width, height = 100, 100, 40, 40
cropped_image = image[y:y+height, x:x+width]
cv2.imshow("Original Image", image)
cv2.waitKey(0)
cv2.imshow("Cropped Image", cropped_image)
cv2.waitKey(0)
cv2.destroyAllWindows()
cv2.imwrite("cropped_image.jpg", cropped_image)

# Rotating
angle = 45
center = (image.shape[1] // 2, image.shape[0] // 2)
rotation_matrix = cv2.getRotationMatrix2D(center, angle, 1.0)
rotated_image = cv2.warpAffine(image, rotation_matrix, (image.shape[1], image.shape[0]))
cv2.imshow("Rotated Image", rotated_image)
cv2.waitKey(0)
cv2.destroyAllWindows()
cv2.imwrite("rotated_image.jpg", rotated_image)
